### ISSUE TITLE
Traversing Clips while in Tab Focus

### ISSUE THREAD

**Issue by chinakov** [audacity/audacity#10384]
### Description

Pressing up/down arrows whilst tab focus is on a clip should move the focus to the nearest clip above/beneath the clip that is in focus rather than moving to the track header.

**Comment by luapmartin** [comment]
@chinakov What is called "Tab Focus" isn't there always something "In Focus" and we just don't show it "always"?

**Comment by DavidBailes** [comment]
> isn't there always something "In Focus" and we just don't show it "always"?

@luapmartin . There should always be a visual indication of which object is currently the keyboard focus. See:
https://www.w3.org/TR/wcag2ict-22/#focus-visible

**Comment by luapmartin** [comment]
yes

**Comment by luapmartin** [comment]
I see another bug @chinakov 
if the "Tab Focus" is on the track header and there is a clip on the track that is visible then tabbing will visually move the "Tab Focus" to that clip, if the clip isn't visible (scrolled timeline viewport) then the tab focus will "stay" for one "tab" on the track header 
I believe the expected behaviour would be to : 
"When going from track header tab focus to track clip focus (if existing) then the timeline viewport should put the clip in view (visually scrolling) and in tab focus"

and that same "put in the view focus update" should also happen for that traversing clip issue I guess as in:

SPECS would then be:

if I move from a clip in focus with UP arrow I would expect to move the focus AND view to the clip (with closest start time/end time/ or central point?) from any track above the current track where clip is in focus if there isn't any track having clip above what should we do?

if I move from a clip in focus with DOWN arrow I would expect to move the focus AND view to the clip (with closest start time/end time/ or central point?) from any track bellow the current track where clip is in focus if there isn't any track having clip bello what should we do?


maybe also LEFT/RIGHT arrow keys should allow moving clip focus+view within one track (circling back to first/last)?

**Comment by luapmartin** [comment]
SPECS UPDATE:
- when focus on track header TAB move focus to closest clip to the current view (no scroll)
- if clip in focus LEFT/RIGHT move focus to prev/next clip AND scroll/adjust view if required
- if clip in focus UP/DOWN move focus to closest clip to the current view (no scroll)

EXTRA BUG:
- WHEN selecting clip (mouse selection) LEFT/RIGHT should move FOCUS but not selection (currently clip wrongly maintain focus)

**Comment by DavidBailes** [comment]
> SPECS UPDATE:
> 
> * when focus on track header TAB move focus to closest clip to the current view (no scroll)
> * if clip in focus LEFT/RIGHT move focus to prev/next clip AND scroll/adjust view if required
> * if clip in focus UP/DOWN move focus to closest clip to the current view (no scroll)

@luapmartin . In the first and third cases, having no scrolling may lead to the object which is the keyboard focus not being visible. It should be visible.

**Comment by teetow** [comment]
I see how that might look implied, but it's not. Rather, we exclude certain things from the focus circle if they're out of view. 

Clarification, though, @luapmartin: We should always focus _something_. If there are no eligible clips according to those rules, we can definitely pick the next best candidate and bring it into view.

**Comment by DavidBailes** [comment]
> I see how that might look implied, but it's not. Rather, we exclude certain things from the focus circle if they're out of view.

@teetow . Please bear in mind whether or not this affects the usability of this for users of screen readers, most of whom won't know what is in or out of view.

**Comment by DilsonsPickles** [comment]
@luapmartin @chinakov @teetow are we referencing the behaviour found in the demo I shared last week?
